Dockerfile

```diff
@@ -9,53 +9,96 @@ ARG CUDA_VERSION=12.2.2
     apt-get update && \
-    apt-get clean
-RUN export DEBIAN_FRONTEND=noninteractive && \
-    apt-get -y install ros-noetic-desktop-full
-RUN export DEBIAN_FRONTEND=noninteractive && \
-    apt-get -y install python3-rosdep python3-rosinstall python3-rosinstall-generator python3-wstool build-essential && \
-    apt-get clean && \
-    rosdep init && \
+    apt-get install -y --no-install-recommends \
+        ros-noetic-desktop-full \
+        python3-rosdep \
+        python3-rosinstall \
+        python3-rosinstall-generator \
+        python3-wstool \
+        build-essential \
+        ros-noetic-catkin \
+        python3-catkin-tools \
+        python3-pip && \
+    # Clean up layer after using apt-get update
+    rm -rf /var/lib/apt/lists/* && apt-get clean
+    
+RUN rosdep init && \
     rosdep update
 
 # Source ROS in all terminals
 RUN echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc
 
-## Stage 3: Install MoveIt 1 for ROS Noetic from source (includes moveit_tutorials and panda_moveit_config)
+## Stage 2: Install MoveIt 1 for ROS Noetic from source (includes moveit_tutorials and panda_moveit_config)
 # Reference: http://moveit.ros.org/install/source/
 FROM noetic AS noetic-moveit
 
-# Install required catkin build tools and Git
-RUN apt-get -y install ros-noetic-catkin python3-catkin-tools git
+# Install Git and clean up afterwards
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends git && \
+    rm -rf /var/lib/apt/lists/* && apt-get clean
 
-# Download MoveIt's source code into a new workspace, install its dependencies, and build
+# Build MoveIt from source in a new workspace within the container
 WORKDIR /moveit_ws
 RUN wstool init src && \
     wstool merge -t src https://raw.githubusercontent.com/moveit/moveit/master/moveit.rosinstall && \
-    wstool update -t src
-RUN rosdep install -y --from-paths src --ignore-src --rosdistro ${ROS_DISTRO}
-RUN catkin config --extend /opt/ros/${ROS_DISTRO} --cmake-args -DCMAKE_BUILD_TYPE=Release
-RUN catkin build
+    wstool update -t src && \
+    rosdep install -y --from-paths src --ignore-src --rosdistro "${ROS_DISTRO}"
+RUN catkin config --extend "/opt/ros/${ROS_DISTRO}" --cmake-args -DCMAKE_BUILD_TYPE=Release && \
+    catkin build
+
+VOLUME /moveit_ws
 
 # Source MoveIt in all terminals
 RUN echo "source /moveit_ws/devel/setup.bash" >> ~/.bashrc
 
-VOLUME /moveit_ws
+# Finalize the default working directory for the image
+ARG DEFAULT_WORKDIR=/spot_skills
+WORKDIR ${DEFAULT_WORKDIR}
+
+## Stage 3: Install the Boston Dynamics Python packages and the Spot SDK
+#   Build on the noetic-moveit image by default (but ARG enables override)
+ARG SPOT_BASE_IMAGE=noetic-moveit
+
+FROM ${SPOT_BASE_IMAGE} AS spot-sdk
+ARG SPOT_SDK_VERSION
+
+# TODO: An issue may arise where the Spot-SDK-only container needs something like:
+#   RUN apt-get install -y python${PYTHON_VERSION} python3-pip
```

Comment:
Edit this: Fix a modern Python version after sanity-checking that it should work.

Reply:
Done. Starting with Python 3.12 fixed.